Content/NPCs/Misc/LootWraith.cs

```diff
@@ -0,0 +1,436 @@
+					}
+
+					Target.velocity -= Target.DirectionTo(NPC.Center) * 15;
+					screechTimer = 0;
+				}
+
+				if (NPC.Distance(Target.Center - NPC.DirectionTo(Target.Center) * 60) > 10)
+					NPC.velocity += NPC.DirectionTo(Target.Center - NPC.DirectionTo(Target.Center) * 60) * 0.2f;
+
+				if (NPC.velocity.Length() > 10)
+				{
+					NPC.velocity.Normalize();
+					NPC.velocity *= 10;
+				}
+
+				NPC.velocity = Vector2.Lerp(NPC.velocity, NPC.DirectionTo(ChainStart), NPC.Distance(ChainStart) * 0.001f);
+
+				Tile tile = Framing.GetTileSafely(xTile, yTile);
+				Chest chest = Main.chest.Where(n => n != null && n.x == xTile && n.y == yTile).FirstOrDefault();
+				if (ChainStart.Distance(Target.Center) < 500 && (!tile.HasTile || chest.frame > 0 || chargeupCounter > 0))
+				{
+					chargeupCounter += 0.01f;
+					if (chargeupCounter >= 1)
+					{
+						if (NPC.velocity == Vector2.Zero)
+						{
+							SoundEngine.PlaySound(SoundID.DeerclopsScream with { Pitch = 0.6f }, NPC.Center);
+						}
+
+						xFrame = 1;
+						NPC.rotation = 0;
+						NPC.velocity.Y = -8;
+						if (NPC.Distance(ChainStart) > 100)
+						{
+							Helper.PlayPitched("Impacts/GlassExplodeShort", 1, Main.rand.NextFloat(0.1f, 0.3f), NPC.Center);
+							chargeupCounter = 0;
+							NPC.damage = 25;
+							NPC.dontTakeDamage = false;
+							chain.useEndPoint = false;
+							enraged = true;
+							NPC.rotation = 0;
+							CameraSystem.shake += 6;
+
+							for (int i = 0; i < 14; i++)
+							{
+								Vector2 dir = Main.rand.NextVector2CircularEdge(1, 1);
+								Dust.NewDustPerfect(NPC.Center + dir * 25, ModContent.DustType<Dusts.GlowLineFast>(), dir * Main.rand.NextFloat(10), 0, Color.Cyan, 1);
+							}
+
+							ChainGores();
+						}
+					}
+					else
+					{
+						Vector2 dustDir = Main.rand.NextVector2CircularEdge(1, 1);
+						Dust.NewDustPerfect(NPC.Center + dustDir * 15, ModContent.DustType<Dusts.Glow>(), dustDir * Main.rand.NextFloat(5), 0, Color.Cyan, chargeupCounter * 0.5f);
+						NPC.rotation = NPC.rotation + Main.rand.NextFloat(chargeupCounter * -0.1f, chargeupCounter * 0.1f);
+						NPC.velocity = Vector2.Zero;
+					}
+				}
+			}
+			else
+			{
+				xFrame = 1;
+
+				if (++chargeCounter % 150 == 0)
+				{
+					takenKnockback = false;
+					chargeVel = NPC.DirectionTo(Target.Center) * 15;
+				}
+
+				chargeVel *= 0.96f;
+
+				if (chargeVel.Length() > 1 && !takenKnockback)
+				{
+					NPC.velocity = chargeVel;
+				}
+				else
+				{
+					Vector2 dir = Main.rand.NextVector2CircularEdge(1, 1);
+					Dust.NewDustPerfect(NPC.Center + new Vector2(5 * NPC.spriteDirection, -6) - dir * 25, ModContent.DustType<GlowLineFast>(), dir * 4, 0, Color.MediumPurple, MathHelper.Min((chargeCounter % 150 - 50) / 100f, 0.7f));
+					NPC.velocity *= 0.96f;
+				}
+			}
+
+			if (xFrame == 1)
+			{
+				oldPos.Add(NPC.Center);
+				if (oldPos.Count > 10)
+					oldPos.RemoveAt(0);
+			}
+
+			if (chain is null || NPC.DistanceSQ(Target.Center) > 4000000 || enraged)
```

Comment:
This seems like a rather large number to check against -- is this arbitrary? Should we set it lower?